pkg/api/node_api.go

```diff
@@ -605,19 +605,43 @@ func (a *NodeApi) nodeProcesses(w http.ResponseWriter, _ *http.Request) error {
 	return nil
 }
 
+func (a *NodeApi) stateHashDebug(height proto.Height) (*proto.StateHashJSDebug, error) {
+	stateHash, err := a.state.StateHashAtHeight(height)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get state hash at height %d", height)
+	}
+	stateHashDebug := proto.NewStateHashJSDebug(*stateHash, height, a.app.version().Version)
+	return &stateHashDebug, nil
+}
+
 func (a *NodeApi) stateHash(w http.ResponseWriter, r *http.Request) error {
 	s := chi.URLParam(r, "height")
 	height, err := strconv.ParseUint(s, 10, 64)
 	if err != nil {
 		// TODO(nickeskov): which error it should send?
 		return &BadRequestError{err}
 	}
+	stateHashDebug, err := a.stateHashDebug(height)
+	if err != nil {
+		return errors.Wrap(err, "failed to get state hash debug")
+	}
 
-	stateHash, err := a.state.StateHashAtHeight(height)
+	if err := trySendJson(w, stateHashDebug); err != nil {
+		return errors.Wrap(err, "stateHash")
+	}
+	return nil
+}
+
+func (a *NodeApi) stateHashLast(w http.ResponseWriter, _ *http.Request) error {
+	height, err := a.state.Height()
+	if err != nil {
+		return errors.Wrap(err, "failed to get the last height")
+	}
+	stateHashDebug, err := a.stateHashDebug(height)
 	if err != nil {
-		return errors.Wrapf(err, "failed to get state hash at height %d", height)
+		return errors.Wrap(err, "failed to get state hash debug")
```

Comment:
Please remove error wrap here or replace message with more meaningful "failed to get last state hash".  From this message we can see at what API path error occured.